test/unittest/dds/xtypes/type_representation/TypeObjectUtilsTests.cpp

```diff
@@ -0,0 +1,2008 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file
+ * This file contains negative tests related to the TypeObjectUtils API.
+ */
+
+#include <gtest/gtest.h>
+
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastdds/dds/xtypes/common.hpp>
+#include <fastdds/dds/xtypes/exception/Exception.hpp>
+#include <fastdds/dds/xtypes/type_representation/TypeObject.h>
+#include <fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp>
+
+#include <fastdds/xtypes/type_representation/BuiltinAnnotations.h>
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+namespace xtypes1_3 {
+
+// Build TypeObjectHashId object with wrong discriminator.
+TEST(TypeObjectUtilsTests, build_type_object_hash_id_wrong_discriminator)
```

Comment:
The idea is to check the negative cases because the positive ones are already checked in Fast DDS-Gen test suite. However, as the positive check is most times trivial, it has been also included. But the test goal is to check the inconsistencies.